src/runtime_src/hip/api/hip_event.cpp

```diff
@@ -7,64 +7,66 @@
 #include "hip/hip_runtime_api.h"
 
 #include "hip/core/event.h"
+#include "hip/core/stream.h"
 
 namespace xrt::core::hip {
 
-static hipEvent_t
-hip_event_create()
+static command_handle hip_event_create()
 {
-  throw std::runtime_error("Not implemented");
+  return insert_in_map(command_cache, std::make_shared<event>(nullptr));;
 }
 
-static void
-hip_event_destroy(hipEvent_t event)
+static void hip_event_destroy(hipEvent_t eve)
 {
-  throw_invalid_value_if(!event, "event passed is nullptr");
-  throw std::runtime_error("Not implemented");
+  throw_invalid_value_if(!eve, "event passed is nullptr");
+  command_cache.remove(eve);
 }
 
-static void
-hip_event_record(hipEvent_t event, hipStream_t stream)
+static void hip_event_record(hipEvent_t eve, hipStream_t stream)
 {
-  throw_invalid_value_if(!event, "event passed is nullptr");
+  throw_invalid_value_if(!eve, "event passed is nullptr");
   throw_invalid_value_if(!stream, "stream passed is nullptr");
-  throw std::runtime_error("Not implemented");
+  /* TODO
+  auto hip_stream = get_stream(stream);
+  auto hip_ev = std::dynamic_pointer_cast<event>(command_cache.get(eve));
+  hip_ev->record(hip_stream);
+  */
 }
 
-static void
-hip_event_synchronize(hipEvent_t event)
+static void hip_event_synchronize(hipEvent_t eve)
 {
- throw_invalid_value_if(!event, "event passed is nullptr");
- throw std::runtime_error("Not implemented");
+  throw_invalid_value_if(!eve, "event passed is nullptr");
+  auto hip_ev = std::dynamic_pointer_cast<event>(command_cache.get(eve));
```

Comment:
So now we have `event`, `eve`, `ev` in the same line...
The change `event` to `eve` does not seem like a great progress.

Reply:
event is the class name and since I wanted to make dynamic_cast_pointer I did not want to use event as the input of function, so I changed it to eve. I can use in_event or something similar.

Reply:
Ah it was no longer compiling then. I thought it was just to get the line shorter.
Whatever naming is good. Thanks.

Reply:
no problem :)